exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -102,6 +125,29 @@ public void setOption(OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, final OptionType type) {
+    try {
+      SystemOptionManager.getValidator(name); // ensure the option exists
+    } catch (final IllegalArgumentException e) {
+      throw UserException.validationError(e)
+        .build(logger);
```

Comment:
if the `getValidator` fails, can the caller actually handle the situation differently from just failing the query ?